recipes/zlib/1.2.11/conanfile.py

```diff
@@ -12,13 +12,20 @@ class ZlibConan(ConanFile):
     description = ("A Massively Spiffy Yet Delicately Unobtrusive Compression Library "
                    "(Also Free, Not to Mention Unencumbered by Patents)")
     settings = "os", "arch", "compiler", "build_type"
-    options = {"shared": [True, False], "fPIC": [True, False], "minizip": [True, False]}
```

Comment:
We can't remove an options, instead, we deprecate them, for [example](https://github.com/conan-io/conan-center-index/blob/5e6089005f0bb66cd16db7b0e5f37f5081c7820c/recipes/dcmtk/all/conanfile.py#L24). It avoids breaking users as the package will be different and also the package behavior.

As minizip is a different package now, it could be a requirement instead, like:


```python
def requirements(self):
    self.options.with_minizip:
        self.requires("minizip/1.2.11")
```

Reply:
That was my initial approach, but then if you try to build it conan refuses. 
There is a cyclical dependency (https://github.com/conan-io/conan-center-index/blob/a5b87478223ded98fea2970a436c5e54961269e0/recipes/minizip/1.2.11/conanfile.py#L33). If we go down that path (i.e. breaking the cycle), the best we can do is to keep the old option and at config time to throw an exception asking from the user to use now the new minizip package. 
If we can not introduce this breaking change, then both packages should remain as is, and sync the recipes periodically. No idea what is better. That is why I open https://github.com/conan-io/conan-center-index/issues/5009 :)

Reply:
So please deprecate that option, but don't raise an exception, instead, follow the usual pattern: https://github.com/conan-io/conan-center-index/pull/5020/files

Reply:
I pushed an extra commit, but I am not 100% certain that it is correct. Even with the deprecation warning, the behavior of the package is changing (at least if we remove the minizip package). Should we keep the minizip in this zlib package too, but with the deprecation warning ?